lib/facter/util/linux/proc.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Facter
+  module Util
+    module Linux
+      class Proc
+        class << self
+          def getenv_for_pid(pid, field)
+            path = "/proc/#{pid}/environ"
+            lines = Facter::Util::FileHelper.safe_readlines(path, [], "\0", chomp: true)
+            lines.each do |line|
+              if line.slice(0, field.length) == field && line.slice(field.length) == '='
+                return line.slice(field.length + 1, line.length - (field.length + 1))
+              end
```

Comment:
`line.slice(0, field.length) == field` seems to be checking if the `line` starts with `field`. Suggest simplifying to:

```
key, value = line.split('=', 2)
return value if key == field
```

which I think gives the desired result in both the "match" and "non-match" cases:

```
irb(main):013:0> key, value = "foo=bar".split('=', 2)
=> ["foo", "bar"]
irb(main):014:0> puts key; puts value
foo
bar
=> nil
irb(main):015:0> key, value = "foobar".split('=', 2)
=> ["foobar"]
irb(main):016:0> puts key; puts value.inspect
foobar
nil
=> nil
```

Reply:
fixed